db: Transaction test race-condition on CI

CI failed unexpectedly in an unrelated PR on `TestTransactionExclusion`: https://circleci.com/gh/0xProject/0x-mesh/696

```
#!/bin/bash -eo pipefail
make test-go

go test ./... -race
?   	github.com/0xProject/0x-mesh/cmd/demo/add_order	[no test files]
?   	github.com/0xProject/0x-mesh/cmd/demo/add_peer	[no test files]
?   	github.com/0xProject/0x-mesh/cmd/demo/subscribe_to_orders	[no test files]
?   	github.com/0xProject/0x-mesh/cmd/mesh	[no test files]
?   	github.com/0xProject/0x-mesh/cmd/mesh-keygen	[no test files]
?   	github.com/0xProject/0x-mesh/constants	[no test files]
?   	github.com/0xProject/0x-mesh/core	[no test files]
--- FAIL: TestTransactionExclusion (0.02s)
    transaction_test.go:222: 
        	Error Trace:	transaction_test.go:222
        	            				asm_amd64.s:1337
        	Error:      	Should be true
        	Test:       	TestTransactionExclusion
        	Messages:   	expected col1 transaction to open before col0 transaction was committed/discarded
FAIL
FAIL	github.com/0xProject/0x-mesh/db	0.157s
ok  	github.com/0xProject/0x-mesh/ethereum	1.302s
ok  	github.com/0xProject/0x-mesh/ethereum/blockwatch	1.330s
?   	github.com/0xProject/0x-mesh/ethereum/wrappers	[no test files]
ok  	github.com/0xProject/0x-mesh/meshdb	1.116s
ok  	github.com/0xProject/0x-mesh/p2p	4.824s
ok  	github.com/0xProject/0x-mesh/rpc	1.124s
ok  	github.com/0xProject/0x-mesh/zeroex	2.026s
ok  	github.com/0xProject/0x-mesh/zeroex/orderwatch	1.163s
Makefile:20: recipe for target 'test-go' failed
make: *** [test-go] Error 1
Exited with code 2
```

--- a/db/global_transaction_test.go
+++ b/db/global_transaction_test.go
@@ -329,7 +329,7 @@ func TestGlobalTransactionExclusion(t *testing.T) {
 		}()
 	}()
 
-	time.Sleep(1 * time.Millisecond)
+	time.Sleep(5 * time.Millisecond)
 	discardSignal <- struct{}{}
 	require.NoError(t, txn.Discard())
 

--- a/db/transaction_test.go
+++ b/db/transaction_test.go
@@ -252,7 +252,7 @@ func TestTransactionExclusion(t *testing.T) {
 	}()
 
 	// A short sleep is necessary to ensure that the goroutines have time to run.
-	time.Sleep(1 * time.Millisecond)
+	time.Sleep(5 * time.Millisecond)
 	discardSignal <- struct{}{}
 	require.NoError(t, txn.Discard())
 

